src/TypeErrorDef.hs

```diff
@@ -0,0 +1,55 @@
+module TypeErrorDef where
```

Comment:
I mentioned this in another comment, but I don't actually see much benefit to separating this out and ignoring orphan instances unless we truly make error handling generic. If our long term plan is to make the type error module's show instance a generic holder for other types, that works, but it feels needlessly complicated to me.

That said this has the major benefit of being significantly easier to read.

Reply:
We need to pull it out because `AssignTypes` and `TypeError` both depend on it, and `TypeError` depends on `AssignTypes`, meaning we would have a circular dependency otherwise.

Reply:
Couldn't we just move `typeVariablesInOrderOfAppearance` into `Types.hs` instead? It seems like a more appropriate home for that function anyway since it only refers to itself and is a generic operation on `Ty` constructors, having nothing to do with the actual assignment of types. And then this would permit us to move `beautifyTys` into typeError or Types.hs as well.

Reply:
Sadly, we can’t move this into `Types` (which I think would be nicest) because it depends on `TypeError`. We also can’t move `beautifyTypeVariables` out of `AssignTypes`, since it depends on `assignTypes`. So we have  to move out `beautifyTy` and `typeVariablesInOrderOfAppearance`. This is a little confusing, but  maybe the best solution available to us.

It also solves the orphan instances problem.

Rectified in `64630d75`!